File: 25 - Balance the Parentheses Count/solution.js
function balanceParentheses(str) {
  // Your code here
  let open = 0;
  let insertion = 0;

  for (char of str) {
    if (char === "(") {
      open++;
    } else {
      if (open > 0) {
        open--;
      } else {
        insertion++;
      }
    }
  }

  return open + insertion;
}

//   Test cases
console.log(balanceParentheses("((()))("));

File: 25 - Balance the Parentheses Count/solution.py
def balance_parentheses(str):
    # Your code here
    open = 0
    insertion = 0

    for char in str:
        if char == '(':
            open += 1
        else:
            if open > 0:
                open -= 1
            else:
                insertion += 1

    return open + insertion


#   Test cases
print(balance_parentheses("((()))("))